src/ui/marker.js

```diff
@@ -790,6 +817,7 @@ export default class Marker extends Evented {
      * const alignment = marker.getPitchAlignment();
      */
     getPitchAlignment(): string {
-        return this._pitchAlignment === `auto` ? this.getRotationAlignment() : this._pitchAlignment;
+        if (this._map && this._pitchAlignment === 'horizon' && !this._map._usingGlobe()) { return 'viewport'; }
+        return this._pitchAlignment === 'auto' ? this.getRotationAlignment() : this._pitchAlignment;
```

Comment:
These checks are starting to become difficult to follow and ready. Partly because `_usingGlobe` is a misleading name. To make it self-documenting, I think we should move away from referring that's it's about _using_ globe and clarify that it's whether the map is actively _showing_, so how about:
1. Renaming `map._usingGlobe` to `map._showingGlobe()`
2. Adding a helper function:
```
_usingHorizonAlignment(): boolean {
    return this._pitchAlignment === 'horizon' && this._map && !this._map._showingGlobe();
}
```
3. Expanding these checks for clarity:
```suggestion
        if (this._usingHorizonAlignment()) {
            return 'viewport';
        }
        if (this._pitchAlignment === 'auto') {
            return this.getRotationAlignment();
        }
        return this._pitchAlignment;
```

Reply:
Good call Karim! I agree that `map._showingGlobe()` is much clearer.